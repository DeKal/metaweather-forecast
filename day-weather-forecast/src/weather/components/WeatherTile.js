import React from 'react'
import PropTypes from 'prop-types'
import { Grid, Typography, Container } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import { format } from 'date-fns'
import theme from 'core/styles/theme'

const useStyles = makeStyles(() => ({
  container: {
    border: theme.borders.standard[1],
    borderWidth: '0 1px',
    margin: `${theme.spacing(2)}px 0`
  },
  dateText: {
    textAlign: 'center',
    marginBottom: theme.spacing(0.5),
    color: theme.colors.hardEmphasis
  },
  maxWeatherText: {
    textAlign: 'center',
    color: theme.colors.primary[1]
  },
  minWeatherText: {
    textAlign: 'center',
    color: theme.colors.primary[2]
  }
}))

const WeatherTile = ({ date, maxTemperature, minTemperature }) => {
  const classes = useStyles()

  const weatherDate = new Date(date)
  const roundedMaxTemperature = Math.round(maxTemperature)
  const roundedMinTemperature = Math.round(minTemperature)
  return (
    <Container maxWidth="xs" className={classes.container}>
      <Grid container direction="column">
        <Grid item className={classes.dateText}>
          <Typography variant="subtitle1">
            {format(weatherDate, 'EEEE d')}
          </Typography>
        </Grid>

        <Grid item className={classes.maxWeatherText}>
          <Typography variant="h4">{`${roundedMaxTemperature}°C`}</Typography>
        </Grid>
        <Grid item className={classes.minWeatherText}>
          <Typography variant="h6">{`${roundedMinTemperature}°C`}</Typography>
        </Grid>
      </Grid>
    </Container>
  )
}

export default WeatherTile
WeatherTile.propTypes = {
  date: PropTypes.string,
  minTemperature: PropTypes.number,
  maxTemperature: PropTypes.number,
  isShowOnlyLeftBorder: PropTypes.bool
}
WeatherTile.defaultProps = {
  date: '20/11/2000',
  minTemperature: 18,
  maxTemperature: 0
}
