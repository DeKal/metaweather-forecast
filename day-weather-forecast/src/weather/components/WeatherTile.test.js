import React from 'react'
import { render } from '@testing-library/react'
import WeatherTile from 'weather/components/WeatherTile'

describe('WeatherTile test', () => {
  test('render correctly with rounded temp', () => {
    const { container } = render(
      <WeatherTile minTemperature={0} maxTemperature={1} date="2021-03-18" />
    )
    expect(container).toMatchSnapshot()
  })
  test('render correctly with floating temp', () => {
    const { container } = render(
      <WeatherTile
        minTemperature={5.6}
        maxTemperature={12.3}
        date="2021-03-18"
      />
    )
    expect(container).toMatchSnapshot()
  })
})
