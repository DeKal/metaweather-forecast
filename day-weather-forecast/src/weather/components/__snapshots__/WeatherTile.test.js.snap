// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherTile test render correctly with floating temp 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-container-5 MuiContainer-maxWidthXs"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root makeStyles-dateText-6 MuiGrid-item"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Thursday 18
        </h6>
      </div>
      <div
        class="MuiGrid-root makeStyles-maxWeatherText-7 MuiGrid-item"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          12째C
        </h4>
      </div>
      <div
        class="MuiGrid-root makeStyles-minWeatherText-8 MuiGrid-item"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          6째C
        </h6>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherTile test render correctly with rounded temp 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthXs"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    >
      <div
        class="MuiGrid-root makeStyles-dateText-2 MuiGrid-item"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          Thursday 18
        </h6>
      </div>
      <div
        class="MuiGrid-root makeStyles-maxWeatherText-3 MuiGrid-item"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          1째C
        </h4>
      </div>
      <div
        class="MuiGrid-root makeStyles-minWeatherText-4 MuiGrid-item"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          0째C
        </h6>
      </div>
    </div>
  </div>
</div>
`;
